File: typescript/src/structural/decorator/models/TextMessage.ts
import { IMessage } from "../interfaces/IMessage";

export class TextMessage implements IMessage {
    private content: string;

    constructor(content: string){
        this.content = content
    }

    processMessage(): IMessage {
        return this
    }

    getContent(): string {
        return this.content
    }

    setContent(content: string): void {
        this.content = content
    }
}

File: c#/Structural/Decorator/Models/TextMessage.cs
using Interface.IMessage;

namespace Models.TextMessage
{
    public class TextMessage : IMessage
    {
        protected string content;

        public TextMessage(string content)
        {
            this.content = content;
        }

        public string GetContent()
        {
            return this.content;
        }

        public IMessage ProcessMessage()
        {
            return this;
        }

        public void SetContent(string content)
        {
            this.content = content;
        }
    }

}
